app/assets/javascripts/discourse/app/components/post/small-action.gjs

```diff
@@ -0,0 +1,206 @@
+import PostCookedHtml from "discourse/components/post/cooked-html";
+import UserAvatar from "discourse/components/user-avatar";
+import concatClass from "discourse/helpers/concat-class";
+import icon from "discourse/helpers/d-icon";
+import { autoUpdatingRelativeAge } from "discourse/lib/formatter";
+import getURL from "discourse/lib/get-url";
+import { applyValueTransformer } from "discourse/lib/transformer";
+import { userPath } from "discourse/lib/url";
+import { i18n } from "discourse-i18n";
+
+// TODO (glimmer-post-stream) remove the export after removing the legacy widget code
+export const GROUP_ACTION_CODES = ["invited_group", "removed_group"];
+const customGroupActionCodes = [];
+
+export const ICONS = {
+  "closed.enabled": "lock",
+  "closed.disabled": "unlock-keyhole",
+  "autoclosed.enabled": "lock",
+  "autoclosed.disabled": "unlock-keyhole",
+  "archived.enabled": "folder",
+  "archived.disabled": "folder-open",
+  "pinned.enabled": "thumbtack",
+  "pinned.disabled": "thumbtack unpinned",
+  "pinned_globally.enabled": "thumbtack",
+  "pinned_globally.disabled": "thumbtack unpinned",
+  "banner.enabled": "thumbtack",
+  "banner.disabled": "thumbtack unpinned",
+  "visible.enabled": "far-eye",
+  "visible.disabled": "far-eye-slash",
+  split_topic: "right-from-bracket",
+  invited_user: "circle-plus",
+  invited_group: "circle-plus",
+  user_left: "circle-minus",
+  removed_user: "circle-minus",
+  removed_group: "circle-minus",
+  public_topic: "comment",
+  open_topic: "comment",
+  private_topic: "envelope",
+  autobumped: "hand-point-right",
+};
+
+export function addGroupPostSmallActionCode(actionCode) {
+  customGroupActionCodes.push(actionCode);
+}
+
+// only for testing purposes
+export function resetGroupPostSmallActionCodes() {
+  customGroupActionCodes.length = 0;
+}
+
+export default class PostSmallAction extends Component {
+  @cached
+  get CustomComponent() {
+    return applyValueTransformer("post-small-action-custom-component", null, {
+      actionCode: this.code,
+      post: this.post,
+    });
+  }
+
+  get additionalClasses() {
+    return applyValueTransformer("post-small-action-class", [], {
+      post: this.args.post,
+    });
+  }
+
+  get code() {
+    return this.args.post.action_code;
+  }
+
+  @cached
+  get createdAt() {
+    return new Date(this.args.post.created_at);
+  }
+
+  get description() {
+    const when = this.createdAt
+      ? autoUpdatingRelativeAge(this.createdAt, {
+          format: "medium-with-ago-and-on",
+        })
+      : "";
+
+    let who = "";
+    if (this.username) {
+      if (this.isGroupAction) {
+        who = `<a class="mention-group" href="/g/${this.username}">@${this.username}</a>`;
+      } else {
+        who = `<a class="mention" href="${userPath(this.username)}">@${
+          this.username
+        }</a>`;
+      }
+    }
+
+    return htmlSafe(
+      i18n(`action_codes.${this.code}`, { who, when, path: this.path })
+    );
+  }
+
+  @cached
+  get icon() {
```

Comment:
This is only accessed once during normal rendering, and it seems reasonably cheap anyway?

```suggestion
  get icon() {
```

Reply:
My concern is a registered transformer perfoming something expensive.